tests/models/svr.py

```diff
@@ -0,0 +1,115 @@
+import math
+import unittest
+import numpy as np
+from numpy.testing import assert_allclose, assert_array_equal
+from scratchml.models.svr import BaseSVR
+from sklearn.svm import SVR as SkSVR
+from ..utils import generate_regression_dataset, repeat
+
+
+class Test_SVR(unittest.TestCase):
+    """
+    Unit test class for the custom SVR implementation.
+    """
+
+    @repeat(3)
+    def test_linear_kernel(self):
+        """
+        Test the custom SVR with linear kernel against Scikit-Learn's SVR.
+        """
+        X, y = generate_regression_dataset(n_samples=200, n_features=5)
+
+        custom_svr = BaseSVR(kernel="linear", C=1.0, epsilon=0.1)
+        sklearn_svr = SkSVR(kernel="linear", C=1.0, epsilon=0.1)
+
+        custom_svr.fit(X, y)
+        sklearn_svr.fit(X, y)
+
+        custom_pred = custom_svr.predict(X)
+        sklearn_pred = sklearn_svr.predict(X)
+
+        custom_score = custom_svr.score(X, y)
+        sklearn_score = sklearn_svr.score(X, y)
+
+        atol = 1e-1
+        assert_allclose(custom_pred, sklearn_pred, atol=atol, rtol=1e-2)
+        assert abs(custom_score - sklearn_score) / abs(sklearn_score) < 0.1
+
```

Comment:
Please, add validations for the parameters as well so we can compare them against Scikit-Learn's implementation. Check out other tests (such as Random Forest and Perceptron) to get an idea of how to do that.